### ISSUE TITLE
[MNG-5898] Multi-module build with ear fails to resolve war in 3.3.3

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7765]
**[Jonathan Radon](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jonmr)** opened **[MNG-5898](https://issues.apache.org/jira/browse/MNG-5898?redirect=false)** and commented

We have a multi-module build that is very large.  With Maven 3.0.4 we run mvn clean test to save time since packaging takes a significant amount of time.

We recently tried to upgrade to Maven 3.3.3 so that we could use some newer plugins.

With Maven 3.3.3, our ear module fails to resolve its war dependency.

This is easily reproducible by using the j2ee archetype.

---

**Affects:** 3.1.1, 3.2.1, 3.2.2, 3.2.3, 3.2.5, 3.3.1, 3.3.3

**Attachments:**
- [example.zip](https://issues.apache.org/jira/secure/attachment/12764844/example.zip) (_5.88 kB_)

**Backported to:** [3.2.6](https://github.com/apache/maven/milestone/81?closed=1)

4 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Jonathan Radon](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jonmr)** commented

I'm attaching an example project generated from the j2ee archetype.

Simply run mvn clean test on it to reproduce.

Works on 3.0.4 and not 3.3.3.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Confirmed. Seems like version 3.1.1 introduced regression.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

So by using git bisect i could identify the culprit

```
52beaf62219327a8b078b198990098e89b88889a is the first bad commit
commit 52beaf62219327a8b078b198990098e89b88889a
...
    MNG-5214: Dependency resolution substitutes g:a:v:jar for j:a:v:something-else when something-else isn't in the reacto

    o When Aether asks the ReactorReader for a file for an artifact, remember to match type and classifier. And if all else
      fails, do not return target/classes if the desired artifact is not of type 'jar'. Arguably, if type or classifier
      are non-default, then we should never return those default paths at all.
    o Add 'dir' format to make it quicker to run a quick test. If everyone hates this I'll revert it.


    git-svn-id: https://svn.apache.org/repos/asf/maven/maven-3/trunk@1212564 13f79535-47bb-0310-9956-ffa450edef68

:040000 040000 47a1593aded05deb8a523f3859bc051887adc9c6 ddf8697882273e58c9ee0b4afef92455704f5246 M	apache-maven
:040000 040000 a6eca16af8d71d774c738b506f37015bdc236483 7a80d22dd2826989fa16f0fc1fdeab89e968c6cf M	maven-core
```

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

I have made fix which looks good based on my test.
Fixed [72e2478ec589e906afa1409c192cee7e33adde96](https://git-wip-us.apache.org/repos/asf?p=maven.git;a=commitdiff;h=72e2478ec589e906afa1409c192cee7e33adde96)
But i would appreciate any review from others dev with more knowledge of the maven core...

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Fixed on Maven 3.2.6 branch with [a791318cdfeef1827992b4303d97d9b036d4355f](https://git-wip-us.apache.org/repos/asf?p=maven.git;a=commitdiff;h=a791318cdfeef1827992b4303d97d9b036d4355f).

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Unfortunately the IT's are not successful...
